lib/lambda_ethereum_consensus/p2p/block_downloader.ex

```diff
@@ -0,0 +1,132 @@
+defmodule LambdaEthereumConsensus.P2P.BlockDownloader do
+  @moduledoc """
+  This module requests blocks from peers.
+  """
+  alias LambdaEthereumConsensus.P2P
+  use GenStage
+
+  @protocol_id "/eth2/beacon_chain/req/beacon_blocks_by_range/2/ssz_snappy"
+
+  @impl true
+  def init([host]) do
+    {:producer, host}
+  end
+
+  @impl true
+  def handle_demand(incoming_demand, host) do
+    blocks =
+      for _ <- 0..incoming_demand do
+        request_block(host) |> wrap_message()
+      end
+
+    {:noreply, blocks, host}
+  end
+
+  defp request_block(host) do
+    # TODO: get missing slots from DB
+    start_slot = 7_270_097 + :rand.uniform(100_000)
```

Comment:
What is this magic number? can we document it and/or create a constant for it ?

Reply:
It's a random number close to the HEAD minus 100K, to mock the slot choosing aspect of the function. This was replaced by a proper algorithm in #228